install.sh

```diff
@@ -137,23 +113,33 @@ main() {
         ;;
     esac
 
-    local _dir="$(ensure get_default_install_dir)"
-    ensure mkdir -p "$_dir"
+    local _temp
+    if ! _temp="$(ensure mktemp -d)"; then
+        # Because the previous command ran in a subshell, we must manually
+        # propagate exit status.
+        exit 1
+    fi
 
-    local _archive="${_dir}/temporal_cli_latest${_ext}"
+    local _archive="${_temp}/temporal_cli_latest${_ext}"
     ensure downloader "$_url" "$_archive" "$_arch"
-    ensure unzip "$_archive" "$_dir"
-    ensure rm "$_archive"
+    ensure unzip "$_archive" "$_temp"
+    local _dir="$(ensure get_install_dir "$@")"
```

Comment:
I think this would need manual exit propagation as well?

Reply:
addressed